cdap-ui/app/features/hydratorplusplus/controllers/create/toppanel-ctrl.js

```diff
@@ -64,8 +45,27 @@ class HydratorPlusPlusTopPanelCtrl{
       this.openMetadata();
     }
 
+    this.isPreviewEnabled = MY_CONFIG.hydrator.previewEnabled === 'true' || MY_CONFIG.hydrator.previewEnabled === true;
```

Comment:
Just to add a line of defense we could probably have a check of ```angular.isObject(MY_CONFIG.hydrator)``` before checking for 'true' or true.